src/commands/reminder/command.ts

```diff
@@ -0,0 +1,79 @@
+import { addMilliseconds, formatDistanceToNow, isBefore } from "date-fns";
+import { TextChannel } from "discord.js";
+import parseDuration from "parse-duration";
+import { client } from "../..";
+import { DI } from "../../database";
+import { Reminder, User } from "../../database/entities";
+import { Command } from "../../types/Command";
+
+let reminderCache: Reminder[] = [];
+
+const reminderInterval = setInterval(() => {
+  const pastReminders = reminderCache.filter((r) => {
+    return isBefore(r.triggerAt, new Date());
+  });
+
+  const futureReminders = reminderCache.filter((r) => {
+    return !isBefore(r.triggerAt, new Date());
+  });
+
+  reminderCache = futureReminders;
+
+  pastReminders.forEach((r) => {
+    const channel = client.channels.cache.get(r.context) as TextChannel;
+    channel.send(r.message);
+
+    DI.reminderRepository.removeAndFlush(r);
+  });
+}, 1000);
+
+async function refreshRemindersCache() {
+  reminderCache = await DI.reminderRepository.findAll();
+}
+
+const command: Command = {
+  name: "Reminder",
+  description: "Reminds you",
+  invocations: ["reminder", "remindme"],
+  args: true,
+  enabled: true,
+  usage: "[invocation]",
+  async onload() {
+    refreshRemindersCache();
+  },
+  async execute(message, args) {
+    if (args.length < 2) return;
+    const duration = args.shift();
```

Comment:
it does not. command usages should be short version `1d5h`. it would be unreliable to parse long form durations with spaces like that. At least not without introducing a delimiter between the duration and message part of the command.

such as:
`.reminder 1 day 5 hours "my message here"`

else how would you reliably parse:
`.reminder 1 day 5 hours`
when the user intends for it to be duration `1 day` with a message of `5 hours`